File: Demo/MNG-Ads-SDK/PagerNativeViewController.m
//
//  PagerNativeViewController.m
//  MNG-Ads-SDK-Demo
//
//  Created by Hussein Dimessi on 1/13/17.
//  Copyright © 2017 MNG. All rights reserved.
//

#import "PagerNativeViewController.h"
#import <BlueStackSDK/MNGNAtiveObject.h>

@interface PagerNativeViewController (){
    MNGAdsSDKFactory *nativeAdsFactory;
    UIView *badgeView;
    UIView *adChoiceBadgeView;
    MNGNAtiveObject *_nativeObject;
}

@end

@implementation PagerNativeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _pubView.hidden = true;
    if (_index == 2) {
        [_activityIndicator startAnimating];
        nativeAdsFactory = [[MNGAdsSDKFactory alloc]init];
        nativeAdsFactory.nativeDelegate = self;
        nativeAdsFactory.viewController = self;
        nativeAdsFactory.placementId = MNG_ADS_NATIVE_AD_PLACEMENT_ID;
        MNGPreference *preferences = [Utils getTestPreferences];
        [nativeAdsFactory loadNativeWithPreferences:preferences];
    }
}

-(void)dealloc{
    [nativeAdsFactory releaseMemory];
    nativeAdsFactory = nil;
}

-(void)viewDidLayoutSubviews{
    [super viewDidLayoutSubviews];
    if (self.view.frame.size.height > self.view.frame.size.width) {
        self.titleLabel.textColor = [UIColor blackColor];
        self.descriptionLabel.textColor = [UIColor blackColor];
    }else{
        self.titleLabel.textColor = [UIColor whiteColor];
        self.descriptionLabel.textColor = [UIColor whiteColor];
    }
}

#pragma mark - MNGAdsAdapterNativeDelegate

-(void)adsAdapter:(MNGAdsAdapter *)adsAdapter nativeObjectDidLoad:(MNGNAtiveObject *)nativeObject{
    [self.pubView.layer setBorderWidth:1];
    [self.pubView.layer setBorderColor:[UIColor lightGrayColor].CGColor];
    self.titleLabel.text = nativeObject.title;
    self.socialContextLabel.text = nativeObject.socialContext;
    //    self.contextLabel.text = nativeObject.socialContext;
    self.descriptionLabel.text = nativeObject.body;
    if (badgeView != nil) {
        [badgeView removeFromSuperview];
    }
    _nativeObject = nativeObject;
    badgeView = _nativeObject.badgeView;
    if (badgeView) {
        [self.badgeContainer addSubview:badgeView];
    }
    
    if (adChoiceBadgeView != nil) {
        [adChoiceBadgeView removeFromSuperview];
    }
    adChoiceBadgeView = _nativeObject.adChoiceBadgeView;
    if (adChoiceBadgeView) {
        
        [self.adChoiceContainer addSubview:adChoiceBadgeView];
    }
    
    
    //download images
    self.backgroundImage.image = nil;
    self.iconImage.image = nil;
    self.iconImage.clipsToBounds = YES;
  
    
    [self.callToActionButton setTitle:_nativeObject.callToAction forState:UIControlStateNormal];
    if(_nativeObject.displayType == MNGDisplayTypeAppInstall){
        [self.callToActionButton setImage:[UIImage imageNamed:@"download"] forState:UIControlStateNormal];
    }else if(_nativeObject.displayType == MNGDisplayTypeContent){
        [self.callToActionButton setImage:[UIImage imageNamed:@"arrow"] forState:UIControlStateNormal];
    }
    self.callToActionButton.titleLabel.textAlignment = NSTextAlignmentCenter;
    [_nativeObject registerViewForInteraction:self.pubView withMediaView:self.backgroundImage withIconImageView:self.iconImage withViewController:[APP_DELEGATE drawerViewController] withClickableView:self.callToActionButton];
    self.pubView.hidden = NO;
    
}

-(void)adsAdapter:(MNGAdsAdapter *)adsAdapter nativeObjectDidFailWithError:(NSError *)error withCover:(BOOL)cover{
    [_activityIndicator stopAnimating];
    NSLog(@"%@",error);
}

@end

File: DemoSwift/DemoSwift/Pager/PagerNativeView/PagerNativeViewController.swift
//
//  PagerNativeViewController.swift
//  BlueStackDemoDev
//
//  Created by MedSghaier on 19/8/2022.
//

import UIKit
import BlueStackSDK

class PagerNativeViewController: UIViewController, MNGAdsAdapterNativeDelegate {
    
    
    @IBOutlet weak var pubView: UIView!
    
    @IBOutlet weak var backgroundImage: UIImageView!
    
    @IBOutlet weak var iconImage: UIImageView!
    
    @IBOutlet weak var titleLabel: UILabel!
    
    @IBOutlet weak var descriptionLabel: UILabel!
    
    @IBOutlet weak var callToActionButton: UIButton!
    
    @IBOutlet weak var badgeContainer: UIView!
    
    @IBOutlet weak var adChoiceContainer: UIView!
    
    @IBOutlet weak var indicatorView: UIActivityIndicatorView!
    
    
    var index : Int = 0
    var nativeAdsFactory : MNGAdsSDKFactory!
    var badgeView : UIView?
    var adChoiceBadgeView : UIView?
    var _nativeObject : MNGNAtiveObject?
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        pubView.isHidden = true
        if index == 2 {
            indicatorView.startAnimating()
            nativeAdsFactory = MNGAdsSDKFactory()
            nativeAdsFactory.nativeDelegate = self
            nativeAdsFactory.viewController = self
            nativeAdsFactory.placementId = "/\(DemoSwiftConstants.appID)/nativead"
            var preferences = self.getTestPreferences()
            nativeAdsFactory.loadNative(withPreferences: preferences)
        }
    }
    
    func getTestPreferences() ->MNGPreference {
        let preference = MNGPreference()
        preference.keyWord = "target=mngadsdemo;version=3.0.4;semantic=https://www.google.com"
        preference.gender = MNGGender.male
        preference.setLocationPreferences(CLLocation.init(latitude: 48.87610, longitude: 10.453), withConsentFlag: 2)
        preference.contentUrl = "your content url"
        preference.age = 60
        return preference
    }


    // MARK: - Native Ad delegate
    func adsAdapter(_ adsAdapter: MNGAdsAdapter!, nativeObjectDidLoad adView: MNGNAtiveObject!) {
        indicatorView.stopAnimating()
        _nativeObject = adView
        self.pubView.layer.borderWidth = 1
        self.titleLabel.text = _nativeObject?.title
        self.descriptionLabel.text = _nativeObject?.body
        
        if badgeView != nil {
            badgeView?.removeFromSuperview()
        }
        badgeView = _nativeObject?.badgeView
        self.badgeContainer.addSubview(badgeView!)
        
        if adChoiceBadgeView != nil {
            adChoiceBadgeView?.removeFromSuperview()
        }
        adChoiceBadgeView = _nativeObject?.adChoiceBadgeView
        if let adChoiceBadgeView = adChoiceBadgeView {
            adChoiceContainer.addSubview(adChoiceBadgeView)
        }
       
        
        // download images
        self.backgroundImage.image = nil
        self.iconImage.image = nil
        self.iconImage.layer.cornerRadius = 16
        self.iconImage.clipsToBounds = true
   
        
        self.callToActionButton.setTitle(_nativeObject!.callToAction, for: UIControl.State())
        if self._nativeObject!.displayType == MNGDisplayType.appInstall {
            self.callToActionButton.setImage(#imageLiteral(resourceName: "download"), for: UIControl.State())
        }else if self._nativeObject!.displayType == .content {
            self.callToActionButton.setImage(#imageLiteral(resourceName: "arrow"), for: UIControl.State())
        }
        self.callToActionButton.titleLabel?.textAlignment = .center
        
        self._nativeObject?.registerView(forInteraction: self.pubView, withMediaView: self.backgroundImage, withIconImageView: self.iconImage, with: self, withClickableView: self.callToActionButton)
        self.pubView.isHidden = false
        
        
    }
    
    func adsAdapter(_ adsAdapter: MNGAdsAdapter!, nativeObjectDidFailWithError error: Error!, withCover cover: Bool) {
        indicatorView.stopAnimating()
        NSLog("\(String(describing: error))")
    }

}
